internal/core/unittest/test_expr.cpp

```diff
@@ -581,3 +581,364 @@ TEST(Expr, TestCompare) {
         }
     }
 }
+
+TEST(Expr, TestBinaryArithOpEvalRange) {
```

Comment:
Please also add ut for expr. As for how to generate plan from expr conviniently, maybe you can refer to `plan_parser_test.go` and `translate_text_plan_to_binary_plan`.

Reply:
Please also add ut for `SegmentInterface.Search` and `SegmentInterface.Retrieve`.
Also, an e2e test demo under `tests/python_client/testcases/` would be perfect.